### PR TITLE
add test for issue 1398 RAI duplicate name

### PR DESCRIPTION
**Pull request by iCollin**
ATF Test Scripts to check https://github.com/smartdevicelink/sdl_core/issues/1398

This PR is **ready** for review.

### Summary
Add new test with cases where RAI should reply with DUPLICATE_NAME

### ATF version
latest

##### Enhancements
* add additional test

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Defects/6_2/1398_RAI_duplicate_name.lua [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        local corId = session:SendRPC("RegisterAppInterface", appParams)
+        session:ExpectResponse(corId, pResult)
+      end)
+  end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL and HMI", common.start)
+runner.Step("Connect six mobile devices to SDL", common.connectMobDevices, {devices})
```
Comment (jacobkeeler) [inline test_scripts/Defects/6_2/1398_RAI_duplicate_name.lua:53]:
```suggestion
runner.Step("Connect mobile device to SDL", common.connectMobDevices, {devices})
```

File: test_scripts/Defects/6_2/1398_RAI_duplicate_name.lua [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+--[[ Local Data ]]
+local devices = {
+  [1] = { host = "1.0.0.1", port = config.mobilePort }
+}
+
+local appParams = {
+  [1] = { appName = "EXISTING1", appID = "0001",  fullAppID = "00000001", vrSynonyms = {"VRSYNONYM1"} },
+  [2] = { appName = "EXISTING2", appID = "0002",  fullAppID = "00000002", vrSynonyms = {"VRSYNONYM2"} },
+  [3] = { appName = "EXISTING3", appID = "0003",  fullAppID = "00000003", vrSynonyms = {"VRSYNONYM3"} },
+  [4] = { appName = "EXISTING1", appID = "0001",  fullAppID = "00000004" },
```
Comment (jacobkeeler) [inline test_scripts/Defects/6_2/1398_RAI_duplicate_name.lua:30]:
```suggestion
  [4] = { appName = "EXISTING1", appID = "0001",  fullAppID = "00000001" },
```

Comment (iCollin) [reply test_scripts/Defects/6_2/1398_RAI_duplicate_name.lua:30]: This case specifies that short appID is the same, not the policyAppId. Such a change would result in APPLICATION_ALREADY_REGISTERED.

Comment (jacobkeeler) [reply test_scripts/Defects/6_2/1398_RAI_duplicate_name.lua:30]: `appID` is exclusively defined as a shortened version of `fullAppID` to account for systems which couldn't handle long app IDs. Back when this issue was introduced, there was no notion of `appID` and `fullAppID`, it used to just be `appID`. 

The case argues that duplicate names should be checked before the app ID, so I believe that it is saying that DUPLICATE_NAME should be returned if both are conflicting. If this is not the case now, then this issue is still technically present. That said, I think nitpicking over the result code in this case is unnecessary, so we can maybe argue that the issue is still invalid.

Comment (iCollin) [reply test_scripts/Defects/6_2/1398_RAI_duplicate_name.lua]: Enlightening. Thank you for explaining.

In https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2403/commits/a6645251bef407988311f115aa40579aa6d5082e I have removed all mention of `fullAppID` for clarity and updated the expectation to be for `APPLICATION_REGISTERED_ALREADY` when both the name and ID of a new app match an existing one.

I consider the other issue to still be fixed as it describes this response as DISALLOWED, this other status is very descriptive and matches the error.
